### PR TITLE
Closes #3149 PricingClient class to get data from pricing endpoint

### LINKED ISSUE #3149: PricingAPIClient class to manage pricing data collection from pricing endpoint
Sub-task for epics #3067 #3068 #3069 

This class will fetch the pricing data from our pricing endpoint and cache it.

### Architecture
- Class will be in `inc/Engine/License/API`, namespace `WP_Rocket\Engine\License\API`
- name is `PricingClient`

#### Methods
- `get_raw_pricing_data()`: the method sends the request to the API endpoint to get the data.
- `get_pricing_data()`: the method first gets the data from the cache (transient `wp_rocket_pricing`), or if no cache, calls `get_raw_pricing_data()` to get it and cache it.

### Tests
- Unit & integration tests for the methods

### PR DESCRIPTION
**Pull request by Tabrisrp**
Closes #3149 

This class is used to get the pricing data using `get_pricing_data()`

If the data is already cached in a transient `wp_rocket_pricing`, it uses the cached data. Else, it sends a request to the API endpoint, and caches the returned JSON data for 6 hours in the transient.

If an error occurs (`WP_Error`, or response code not `200`), the method returns false.

### CODE REVIEW

File: inc/Engine/License/API/PricingClient.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+	private function get_raw_pricing_data() {
+		$response = wp_safe_remote_get(
+			self::PRICING_ENDPOINT
+		);
+
+		if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			return false;
+		}
+
+		return json_decode( wp_remote_retrieve_body( $response ) );
```
Comment (hellofromtonya) [inline inc/Engine/License/API/PricingClient.php:51]: @Tabrisrp `wp_remote_retrieve_body` can return an `array` or `WP_Error`. What if error or nothing is returned?

File: inc/Engine/License/ServiceProvider.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+		'pricing_client',
+	];
+
+	/**
+	 * Registers items with the container
+	 *
+	 * @return void
+	 */
+	public function register() {
+		$views = __DIR__ . '/views';
```
Comment (hellofromtonya) [inline inc/Engine/License/ServiceProvider.php:29]: `$views` isn't used.

File: tests/Unit/inc/Engine/License/API/PricingClient/getPricingData.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+			Functions\expect( 'get_transient' )
+				->once()
+				->with( 'wp_rocket_pricing' )
+				->andReturn( $expected );
+		} else {
+			Functions\expect( 'get_transient' )
+				->once()
+				->with( 'wp_rocket_pricing' )
+				->andReturn( false );
+		}
```
Comment (hellofromtonya) [inline tests/Unit/inc/Engine/License/API/PricingClient/getPricingData.php:32]: Refactor to simplify:

```suggestion
		Functions\expect( 'get_transient' )
			->once()
			->with( 'wp_rocket_pricing' )
			->andReturn( $true === $config['transient'] ? $expected : false );
```

File: tests/Unit/inc/Engine/License/API/PricingClient/getPricingData.php [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+				Functions\when( 'wp_remote_retrieve_response_code' )
+				->justReturn( '' );
+			} elseif (
+				is_array( $config['response'] )
+				&&
+				isset( $config['response']['code'] )
+			) {
+				Functions\when( 'wp_remote_retrieve_response_code' )
+				->justReturn( $config['response']['code'] );
+			}
```
Comment (hellofromtonya) [inline tests/Unit/inc/Engine/License/API/PricingClient/getPricingData.php:50]: Simplify:

```suggestion
		Functions\when( 'wp_remote_retrieve_response_code' )
			->justReturn( 
				is_array( $config['response'] ) && isset( $config['response']['code'] )
				? $config['response']['code'] 
				: ''
			);
```

**Review by hellofromtonya — CHANGES_REQUESTED** [review]

File: inc/Engine/License/ServiceProvider.php (hunk 2 above)
Comment (Tabrisrp) [reply inc/Engine/License/ServiceProvider.php]: Right, it's for later, but I removed it from this PR for now

File: tests/Unit/inc/Engine/License/API/PricingClient/getPricingData.php (hunk 4 above)
Comment (Tabrisrp) [reply tests/Unit/inc/Engine/License/API/PricingClient/getPricingData.php]: Updated too

**Review by hellofromtonya — APPROVED** [review]

**Pull request merged**